src/event.rs

```diff
@@ -0,0 +1,623 @@
+///
+/// [`LdkLite`]: [`crate::LdkLite`]
+#[derive(Debug, Clone)]
+pub enum LdkLiteEvent {
+	/// A payment we sent was successful.
+	PaymentSuccessful(PaymentSuccessfulEvent),
+	/// A payment we sent has failed.
+	PaymentFailed(PaymentFailedEvent),
+	/// A payment has been received.
+	PaymentReceived(PaymentReceivedEvent),
+	// TODO: Implement after a corresponding LDK event is added.
+	//ChannelOpened(ChannelOpenedEvent),
+	/// A channel has been closed.
+	ChannelClosed(ChannelClosedEvent),
+	// TODO: Implement on-chain events when better integrating with BDK wallet sync.
+	//OnChainPaymentSent(OnChainPaymentSentEvent),
+	//OnChainPaymentReceived(OnChainPaymentReceivedEvent),
+}
+
+trait EventType {
+	const TYPE: u8;
+}
+
+impl Readable for LdkLiteEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		match Readable::read(reader)? {
+			PaymentSuccessfulEvent::TYPE => Ok(Self::PaymentSuccessful(Readable::read(reader)?)),
+			PaymentFailedEvent::TYPE => Ok(Self::PaymentFailed(Readable::read(reader)?)),
+			PaymentReceivedEvent::TYPE => Ok(Self::PaymentReceived(Readable::read(reader)?)),
+			// ChannelOpenedEvent::TYPE => {
+			// TODO ChannelOpened
+			//}
+			ChannelClosedEvent::TYPE => Ok(Self::ChannelClosed(Readable::read(reader)?)),
+			//5u8 => {
+			// TODO OnChainPaymentSent
+			//}
+			//6u8 => {
+			// TODO OnChainPaymentReceived
+			//}
+			_ => Err(lightning::ln::msgs::DecodeError::InvalidValue),
+		}
+	}
+}
+
+impl Writeable for LdkLiteEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		match self {
+			Self::PaymentSuccessful(event) => {
+				event.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentFailed(event) => {
+				event.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentReceived(event) => {
+				event.write(writer)?;
+				Ok(())
+			}
+			//Self::ChannelOpened { .. } => {
+			//TODO
+			//}
+			Self::ChannelClosed(event) => {
+				event.write(writer)?;
+				Ok(())
+			} //Self::OnChainPaymentSent { .. } => {
+			  //TODO
+			  //}
+			  //Self::OnChainPaymentReceived { .. } => {
+			  //TODO
+			  //}
+		}
+	}
+}
+
+/// A payment we sent was successful.
+#[derive(Debug, Clone)]
+pub struct PaymentSuccessfulEvent {
+	/// The hash of the payment.
+	pub payment_hash: PaymentHash,
+	/// The wrapped LDK event.
+	pub inner: ldk_events::Event,
+}
+
+impl EventType for PaymentSuccessfulEvent {
+	const TYPE: u8 = 0u8;
+}
+
+impl Readable for PaymentSuccessfulEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		let payment_hash: PaymentHash = Readable::read(reader)?;
+		let inner: ldk_events::Event = MaybeReadable::read(reader)?.unwrap();
+		Ok(Self { payment_hash, inner })
+	}
+}
+
+impl Writeable for PaymentSuccessfulEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		Self::TYPE.write(writer)?;
```

Comment:
Should avoid writing a type in `Writeable` if not reading it in `Readable`. See https://github.com/lightningdevkit/rust-lightning/pull/1749#discussion_r987266950.

Reply:
Will adopt `write_event` here also.